client/src/screens/training/controller.js

```diff
@@ -0,0 +1,45 @@
+export const controller = (serviceLocator) => {
+  const { getStoreData } = serviceLocator.getStore();
+  const { navigationApi } = serviceLocator.getAPIs();
+
+  const getData = () => getStoreData(controller.storeDataAccessors);
+  const getParams = () => navigationApi.getPathParams(
+    navigationApi.routes.openTraining,
+  );
+  
+  return {
+    getTraining: () => {
+      const params = getParams();
+      const trainings = getData().trainings;
+      const training = trainings.find(training => training.id === params.training);
+      return training;
+    },
+    getSets: (exercise) => {
+      const ex = exercise;
+      let sets = '';
+      console.log('bla: ', ex);
+      ex.sets.map((set, index) => {
```

Comment:
`.map()` returns a new array with fixed items. you don't use its return value, but need only to iterate items of the array. this way it's better to use `.forEach`